lib/Dialect/HW/ModuleImplementation.cpp

```diff
@@ -241,6 +241,21 @@ ParseResult module_like_impl::parseModuleFunctionSignature(
 // New Style
 ////////////////////////////////////////////////////////////////////////////////
 
+/// Parse an optional keyword or string and set instance into 'result'.`
+ParseResult parseOptionalKeywordOrOptionalString(OpAsmParser &p,
+                                                 std::string *result) {
+  StringRef keyword;
+  if (succeeded(p.parseOptionalKeyword(&keyword))) {
+    *result = keyword.str();
+    return success();
+  }
+
+  // Just as easy to make the required return check happy this way.
+  if (succeeded(p.parseOptionalString(result)))
+    return success();
+  return success();
```

Comment:
Slightly confused by this comment and the double `return success()`. Is the second supposed to be `return failure()`?

Reply:
I think the parse functions are marked return check required for clang-tidy.  So either explicitly ignore the return or do this.